quickwit/quickwit-search/src/collector.rs

```diff
@@ -170,7 +177,7 @@ pub struct QuickwitSegmentCollector {
     max_hits: usize,
     segment_ord: u32,
     timestamp_filter_opt: Option<TimestampFilter>,
-    aggregation: Option<AggregationSegmentCollector>,
+    aggregation: Option<AggregationSegmentCollectors>,
```

Comment:
Instead of the enum approach, have you considered adding a field:
```
aggregation: Option<AggregationSegmentCollector>,
find_trace_ids: Option<FindTraceids>,
```


```
if let Some() = self.aggregation.as_mut() {
   // ...
}
if let Some() = self.find_trace_ids.as_mut() {
   // ...
}
```

It would have the benefit of not mixing the results.

Reply:
I did not. When I first started, I was hoping that tantivy aggregations also implement `Collector/SegmentCollect`. Unfortunately, I ended up in this ugly situation when I realized too late that it was not the case. 

Since we're about to introduce a new query AST that also carries collectors, I will leave it as is for now.